app/config/api.py

```diff
@@ -29,35 +38,75 @@ def validation_error_to_response(request: HttpRequest, exception: ValidationErro
     messages = extract_error_messages(exception)
     return api.create_response(
         request,
-        {"detail": messages},
+        {
+            "code": status, "description": messages
+        },
         status=status,
     )
 
 
 @api.exception_handler(Http404)
-def http404_to_response(request: HttpRequest, exception: Http404) -> HttpResponse:
-    """Convert the given exception to a Not Found response with the actual exception message.
-
-    Without this, we would only see "Not found" in the response even if the
-    original exception message is more detailed like "No User matches the given query.".
-    """
+def handle_404_not_found(request: HttpRequest, exception: Http404) -> HttpResponse:
     return api.create_response(
         request,
-        {"detail": str(exception)},
+        {
+            "code": 404, "description": "Resource not found"
+        },
         status=404,
     )
 
 
 @api.exception_handler(Exception)
-def exception_to_response(request: HttpRequest, exception: Exception) -> HttpResponse:
-    """Convert the given exception to a generic internal server error."""
+def handle_exception(request: HttpRequest, exception: Exception) -> HttpResponse:
+    logger.exception(exception)
     return api.create_response(
         request,
-        {"detail": str(exception)},
+        {
+            "code": 500, "description": "Internal Server Error"
+        },
         status=500,
     )
 
 
+@api.exception_handler(HttpError)
+def handle_http_error(request: HttpRequest, exception: HttpError) -> HttpResponse:
+    return api.create_response(
+        request,
+        {
+            "code": exception.status_code, "description": exception.message
+        },
+        status=exception.status_code,
+    )
+
+
+@api.exception_handler(AuthenticationError)
+def handle_unauthorized(request: HttpRequest, exception: AuthenticationError) -> HttpResponse:
+    logger.exception(exception)
+    return api.create_response(
+        request,
+        {
+            "code": 401, "description": "Unauthorized"
+        },
+        status=401,
+    )
+
+
+@api.exception_handler(NinjaValidationError)
+def handle_ninja_validation_error(
+    request: HttpRequest, exception: NinjaValidationError
+) -> HttpResponse:
+    messages: list[str] = []
+    for error in exception.errors:
+        messages.extend(error.values())
+    return api.create_response(
+        request,
+        {
+            "code": 422, "description": messages
```

Comment:
Apparently 422 is for WebDAV. Maybe 400 would be better here, it's also more widely used. But there's a whole range of dicussions around that [here](https://stackoverflow.com/questions/16133923/400-vs-422-response-to-post-of-data)

Reply:
Let's discuss and make a decision today.

Reply:
As discussed in person: keep 422 for now.